### ISSUE TITLE
Error running drupal playbook directly on Ubuntu 20.04

### ISSUE THREAD

**Issue by staticdev** [geerlingguy/ansible-for-devops#364]
I am using ansible 2.10.3, and running the playbook with the command:

```sh
ansible-playbook -c local -i 'localhost,' playbook.yml
```

Then I get:

```
PLAY [all] ************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ************************************************************************************************************************************************************************************
[WARNING]: Unhandled error in Python interpreter discovery for host localhost: '<' not supported between instances of 'str' and 'int'
fatal: [localhost]: FAILED! => {"ansible_facts": {}, "changed": false, "failed_modules": {"ansible.legacy.setup": {"ansible_facts": {"discovered_interpreter_python": "/home/static/.pyenv/shims/python3.7"}, "failed": true, "module_stderr": "sudo: a password is required\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1, "warnings": ["Platform linux on host localhost is using the discovered Python interpreter at /home/static/.pyenv/shims/python3.7, but future installation of another Python interpreter could change the meaning of that path. See https://docs.ansible.com/ansible/2.10/reference_appendices/interpreter_discovery.html for more information."]}}, "msg": "The following modules failed to execute: ansible.legacy.setup\n"}

PLAY RECAP ************************************************************************************************************************************************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
```

After reading the mentioned documentation I couldn't really figure out what was the problem here.

BTW @geerlingguy congratulations for the great work, keep on!

**Comment by geerlingguy** [comment]
It looks like if you're trying to run this against localhost—e.g. your local workstation—you would need to provide a sudo password so the playbook can configure everything.

I would recommend running the example against a VM or something first, before applying all the changes locally, if you're running from your normal workstation.

But in any case, to provide a sudo password, you can use the `-k` option.

**Comment by staticdev** [comment]
My mistake, sorry for forgetting that. But also didn't work, command with sudo pass:

```sh
ansible-playbook -c local -i 'localhost,' playbook.yml -K  # also same output with -k
```

Output:

```
ansible-playbook -c local -i 'localhost,' playbook.yml -K
BECOME password: 

PLAY [all] ************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ************************************************************************************************************************************************************************************
[WARNING]: Unhandled error in Python interpreter discovery for host localhost: '<' not supported between instances of 'str' and 'int'
[WARNING]: Platform linux on host localhost is using the discovered Python interpreter at /home/static/.pyenv/shims/python3.7, but future installation of another Python interpreter could change the
meaning of that path. See https://docs.ansible.com/ansible/2.10/reference_appendices/interpreter_discovery.html for more information.
ok: [localhost]

TASK [Update apt cache if needed.] ************************************************************************************************************************************************************************
[WARNING]: Updating cache and auto-installing missing dependency: python3-apt
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Could not import python modules: apt, apt_pkg. Please install python3-apt package."}

PLAY RECAP ************************************************************************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
```

**Comment by geerlingguy** [comment]
It looks like the new error is:

```
Could not import python modules: apt, apt_pkg. Please install python3-apt package.
```

Try installing that and see what happens.

**Comment by staticdev** [comment]
@geerlingguy already installed on root and also on the user I am using it. Very odd.

**Comment by geerlingguy** [comment]
@staticdev - It sounds like it could be a Python environment issue, either if you're using `pyenv`, or if you have multiple instances of Python installed (this happens to me often on my Mac...). I would suggest asking about it in #ansible on Freenode IRC, or in the Ansible Project mailing list, as I'm not the expert on python environment issues :O

**Comment by staticdev** [comment]
@geerlingguy I did what you told me. I were correct, I installed Python 3.9 with pyenv and Ubuntu 20.04 cames with 3.8. The problem is that the installed `python3-apt` just works on the system installed python (you check it with `import apt`).

The "solution" is adding the path to original system interpreter to override the one detected by ansible with `ansible_python_interpreter=/usr/bin/python3.8`. Anyhow, I also think there is a problem with ansible 2.10 either with the detection of the used interpreter or, at least, this message error telling me to install `python3-apt`.
